### ISSUE TITLE
Is possible a Fx Smooth node?

### ISSUE THREAD

**Issue by F-Burning** [opentoonz/opentoonz#3863]
Could be possible to create a Fx copying and pasting the code of this option?

![image](https://user-images.githubusercontent.com/74098734/113578275-75868b80-961a-11eb-9b4b-4f3d9f7340e8.png)

This would work like OLM plugins

![image](https://user-images.githubusercontent.com/74098734/113578424-b7afcd00-961a-11eb-94a9-36ea344ba7ce.png)

Why create a node like this?

-Because you could add a simple node instead go level by level change each configuration
-Because apply the smooth before the compositing is like building the house from the roof
-Because you could add several smooth to a compositing and get a better control of jaggy

**Comment by WolfInABowl** [comment]
I'd love an anti-aliasing node, there are many FX techniques that purposefully create aliased edges then smooth them out. 
Also, some nodes only produce aliased results such as `Posterize`, you need to render an image sequence to have the ability to smooth it out. You could resort to using `Blur`, but that doesn't do a very good job.


On a related note:
After Effects is the industry standard compositing software used in the anime industry; it works on a much more granular level as it is designed to work with regular pixel data. There are plugins to blur colors together, as well as numerous other useful things such as generating lines between color borders. These are applied to aliased images which are then smoothed via anti-aliasing.
Compared to OpenToonz, this allows for a much greater degree of creativity as you are able to work with dynamically generated image data, rather than being restrained to the palette defined in the level. 
E.g. you can easily use `PaletteFilter` to key out areas of color, but if you wanted to dynamically generate new areas with new colors, then `PaletteFilter` is unusable for that purpose.

As OpenToonz applies its anti-aliasing *before* any FX nodes, effects designed to work in the same manner as they do in After Effects would not be possible as they need to work on aliased images. 
The addition of an anti-aliasing node would open the door for future nodes designed to work on any kind of image instead of just ToonzRaster/ToonzVector as the user can choose when the anti-aliasing is applied.

**Comment by andeon** [comment]
I agree that this workflow is much more cleaner, makes more sense for many other types of work or with slightly more complex scenes.

The Fx mode could be good way since you don't need to change the workflow where the Level is used, as I believe that many artists already use this process for anime:

https://youtu.be/lTwMjbD2adQ?t=1247
![Dong Chang - Head Turn](https://i.ytimg.com/vi/lTwMjbD2adQ/maxresdefault.jpg)

**Comment by F-Burning** [comment]
Exactly @WolfInABowl 

Thank you for explain the context, because is difficult to me express in English 😄 

With the smooth node, the RGB key takes more sense, because with this one you can keep or delete any color of an external aliasing drawing, but then you need an antialiasing fx to smooth the lines. This simple node can make the difference, because open the door to the fx compositing with any format, like After effects.


> The Fx mode could be good way since you don't need to change the workflow where the Level is used, as I believe that many artists already use this process for anime:

That is the crux of the matter.

Something as simple as being able to smooth lines, is one of the pillars of Japanese digital animation, and if you can work comfortably with it, I have no doubt that many artists will move to Opentoonz and give the credit that the software deserves.
I'm a freelance animator, and I have been a lot of years working with AE, but now I'm trying to make the move to Opentoonz, and that is the reason why I suggest many things, because I see that it has enormous potential, but little things need to be added or changed for the software to be seriously considered, in my opinion.

**Comment by F-Burning** [comment]
I invoke to @shun-iwasawa to see if it is possible to have this in the near future 🙏

**Comment by RodneyBaker** [comment]
@F-Burning  Heck I can answer that.  Of course it is possible.  ;)
I'm curious to know what you define as 'near future' though.

You may be aware that no additional features are expected to be added this side of the release of v1.5.
That likely takes us at least to the end of April before more features are merged.
Developers are already filling up the cue to add new code to the nightly releases after initial release of v1.5 so they can even now submit a PR for an FX Smooth node for use in the Schematic should there be sufficient interest.  Shun's time is largely being consumed by the v1.5 release however.

So by near future do you mean to suggest the fx smooth feature should be included in v1.6 in the early 2022 timeframe?  ;)

At any rate, I hope you are one of those that lean forward and champion use of the nightly releases.
That  is where the FX Smooth node is most likely to appear.

There are also several active forks of Opentoonz that see a lot of development in the middle of the year.
The Tahoma2D fork just had an Alpha release for instance.
It may be that your desired feature might make an appearance in one of the forks and then be ported back into the Opentoonz master.

**Comment by gab3d** [comment]
@RodneyBaker 
In all honesty I feel OT should have a semestral release, better than an annual one.
1 year is a lot of time, especially considering important bug fixes that usually happen only in nightlies until the next full release.

I know it seems to be more work to be done, but I also think that more frequent releases should lessen the stress load that is now behind "the single release of the entire year". 
Currently there's a somewhat dramatic feel of urge to include things before a release, because if not, then it'll pass a full year until they're in.
I think two releases per year should calm that down a lot, deliver timely bug fixes and not necessarily mean that much extra work.

Just my feelings on this...

**Comment by RodneyBaker** [comment]
I have much to say on this subject but perhaps this is not the ideal place to say it.

Bottom line:  We (collectively) do not control the release schedule but we can leverage the heck out of it.
I see the interim between Opentoonz releases as the time for the forks of Opentoonz to really shine.
Individual developers know best what they need for production... for their users... for their own personal goals in life.
At any time developers can port code from the forks to the master and set the next release up for success.
That could be one year... six months... quarterly... nightly!

I have almost given up championing the nightlies (just kidding... that isn't going to happen!).
The nightlies are almost always more stable than the annual release because they contain the bug fixes that users report.
The primary exception to this is as we near any full release where things tend to slip through the cracks and generate quick turn around patches.  Throughout the year the nightlies are quite stable so in the nightlies we have something even better than semi-annual releases...  we have a release with every merged pull request (PR).  This is where education comes to play... the average user is conditioned by other software to expect interim updates to be less stable.  And the term 'nightly' is a misnomer... sometimes we have more than one release a day... they just get published in a category called 'nightlies'.

Can we release semi-annually.  Sure.
But not without help.
At present we (collectively) are not up to this task.
Perhaps we can rise to the challenge by first recognizing the importance of the automated builds (via Github Actions, appveyor) as well as the nightlies.  Don't like the term 'nightlies'... maybe the workflow releases are a better option.  Those get generated with every PR even before any build is created via the nightlies.

We are not leveraging the services and tools we currently have.
Perhaps if we can we can get to the point of more frequent releases that have an official stamp of approval that users recognize as stable.

There have been over 500 release of Opentoonz since the first release of v1.4
With very few exceptions I personally have used all of them.
And things are set to get even better... perhaps even blasting the idea of annual/semi-annual and other longer term frequencies to bits.

I do think however that before planning too much for the next release we need to lock down this one.  
It's not too early to start working on v1.6 though and there are many ways to do just that.  :)

Semi-annual releases?
Too easy.  You need to thing bigger! :)

**Comment by gab3d** [comment]
@RodneyBaker 
I know the feeling, I also love the nightlies.
I also don't think either that 2 releases per year is somewhat an optimal number of releases to make, in theory more could be even better.

But, on the other hand, I reckon that for "common people" (the not specially tech savvy) or studios that make decisions based on certain standard procedures, official releases are the only versions they really seriously consider (or even, are aware of altogether).

So, with that in mind, having to wait a full year to see a new official version, often ends up giving the impression that the software isn't being actively maintained (in fact I've seen lately quite a lot of people on social media asking if OT was still being developed or if it was superseded by Tahoma2D).

So, as I'm also aware that it's not easy to send new official releases out to the world, I suppose that (at the very least) having an intermediate release mid-year could alleviate those issues, without imposing too much work for the current (limited) organization OT has behind. 
Perhaps that's not possible, and it's alright, I'm grateful for what we do have, but that's how I see things currently regarding the release cycle.

**Comment by F-Burning** [comment]
Sure @RodneyBaker, I mean to implement it in the nightly versions, since I am aware of them almost daily 👍 

I invoked this thread to Shun because he has created custom nodes for the software, so I wanted to ask him personally if it was possible to him, but if you say that it is viable, it only remains to wait for someone to see this thread and find the idea interesting, I suppose. I just hope it is not forgotten.

PD: You will excuse me if I do not enter in the discussion above, but I would like to focus on the main topic beyond other topics, 
because I really consider that this node would be liked by many artists, and why not, by companies as well 😄

**Comment by Mark11110** [comment]
@RodneyBaker
you may want to encourage your management to imitate https://fund.blender.org/ 
Not only could this solve your manpower issues, but when presented correctly could also make your management think highly of you .:wink:
Increased funding is always a good thing for a company.

**Comment by shun-iwasawa** [comment]
As far as I can imagine the major concern for introducing the smooth fx would be that it will work as expected only when the camera dpi is the same as the level's dpi.
If not, the level image will be resized on rendering. Resizing causes antialiasing due to resampling even if the `Add Antialiasing` option is not set in the level settings.

**Comment by F-Burning** [comment]
As long as it is specified in the manual there will be no problem @shun-iwasawa 😄 

I show you an example:

Here I used the calligraphy node after applying the anti-aliasing.

![image](https://user-images.githubusercontent.com/74098734/115524471-149bbc00-a286-11eb-9798-e723e8afc6c8.png)

If you notice, the line appears jagged and not smooth, and the result is not clean at all.

**Comment by F-Burning** [comment]
> So by near future do you mean to suggest the fx smooth feature should be included in v1.6 in the early 2022 timeframe? ;)

👀

**Comment by F-Burning** [comment]
So, could this be possible @RodneyBaker?

At least we can try it, and I think a lot of people will definitive jump from after effects to opentoonz!

**Comment by RodneyBaker** [comment]
Just about anything is possible.
I don't know of anyone currently working on a smooth node for FX.
We can certainly discuss what success will look like and talk of specifics.

As you hae suggested, antialiasing is gained pretty much automatically in many ways inside Opentoonz even to the point where users have suggested they need additional ways to prevent antialising... presumably to get to the crisper graphics seen in low bitmap resolution games etc.

One approach that would be easier than most to getting an antialiasing FX into the Schematic would be to determine if there is currently any Shader examples that interact with antialiasing as that would not only demonstrate the capability but due to licensing the actual code could be used and modified as well.

I do think if there is an antialiasing FX it likely should have more options than adding antialiasing.
It should also have options to limit antialiasing as well as increase it and perform other processes.
This is beyond the scope of your current request but thinking 'future' is part of the process too.

Bottom line:  Yes, it is posslble but no one is currently taking on the challenge.  At least they have your request here to consider.

**Comment by RodneyBaker** [comment]
Here's some GL shader code example:

https://www.geeks3d.com/20110405/fxaa-fast-approximate-anti-aliasing-demo-glsl-opengl-test-radeon-geforce/2/

**Comment by RodneyBaker** [comment]
I should also add that when first seeing this report I assumed it related to vector line smoothing.
I mention this because we'd want to consider the name of the antialiasing node to make sure it conveys the right process.

**Comment by RodneyBaker** [comment]
I should add this as well:

> ...a lot of people will definitive jump from after effects to opentoonz...

Nothing I do is designed to move anyone from after effects (or any other software) to Opentoonz.
That simply is not a primary motivator for me personally.  I do recognize of course that others might wish to move to Opentoonz from other software and that there is a tendency to desire one progarm to perform everything.  This is not ideal in many respects but I do recognize that desire on the part of many users.  For my part I try to think in terms of pipeline... even for one person studios.  That and production driven requirements.

I can't speak to developers approaches but it has been stated by some that their primary motivators for taking on a requests is variously:  1) For the challenge 2) Out of frustration to resolve a problem.  In these cases we mostly then need to demonstrate how one or both of these apply to adding of an antialiasing node.   Shun suggested one of the problems that needs to be considered in adding an antialiasing node.  Perhaps some mechanism could be created that only allows the antialiasing node to be added to collapsed subxsheets or after the xsheet node (or similar stable point) so that the obstacle mentioned can be avoided.  Shun is certainly capable of implementing the antialiasing node you request but as you can see he's also carefully considering the bigger picture and related issues of the future.

**Comment by F-Burning** [comment]
Oh, I see.

I think the antialiasing node could be similar to this panel:

![113578275-75868b80-961a-11eb-9b4b-4f3d9f7340e8](https://user-images.githubusercontent.com/74098734/155011743-7e36b3a2-5a63-4e97-818c-9e5297b64b5f.png)

I mean, considering that the code of what I ask already exists inside opentoonz, do a "copy and paste" into a node fx (speaking lightly 🤣), but I understand that this suggestion has not aroused the interest of the developers and that it is not as simple as I propose. I also thought that being able to use anti-aliasing on collapsed scenes is a good option, as you mentioned.

And sorry, maybe I focused it wrong. What i'm going to is that Opentoonz is a real alternative for Japanese anime production, because the potential is there, but there are little details like this that don't allow you to make the jump from After Effects, and it's a pity, because while Opentoonz is free, After Effects is a constant waste of money (eternal payment subscription), at least if you want to do this professionally 😅 In any case, my intention is not to make "emotional blackmail" with it, far from it, but simply to make known this dilemma that I saw when I found the program, and hence the reason for these suggestions!

Thank you very much for your answer.

**Comment by shun-iwasawa** [comment]
Well, I'm thinking of two ideas, the Smooth fx and an alternative.

1. Introducing the Smooth Fx 
    As I wrote before,using the Smooth Fx will be very limited. Because you cannot apply the fx to any level with camera work.
    Practical usage would be collapsing the level + smooth fx into a subxsheet and apply camera work to the subxsheet.
    It seems to be somewhat consistent with the usage in AfterEffects, but I think it is a bit troublesome...
    ![idea1_smoothfx](https://user-images.githubusercontent.com/17974955/155056164-a2f68470-8663-4b4d-bb56-be29d1197752.jpg)

2. Alternative idea: Raster Palette Filter Fx
    This idea is premised on that you would like to do **RGB-based keying** before smoothing.
    This effect would be specialized to use connecting just after the raster level column. It works like RGB Key Fx, but it will be applied BEFORE antialiased or resampled. (We already have this type of Fx for Toonz levels : the Palette Filter Fx.) 
    ![idea2_rasterpalettefilter](https://user-images.githubusercontent.com/17974955/155056276-118f80d6-0d62-43fa-9764-876a93cf3ed4.jpg)

I prefer the latter idea, as it can keep the scene simple.

**Comment by F-Burning** [comment]
That sounds so good, I love it!!!

I have a important quesiton...this could work in collapsing nodes? In After effects you can pre-compose several layers, and then apply the RGB Keying to the pre-compose (like collapsing in opentoonz), so you can apply the RGB to several layers at once. I think this is not a big problem, but would make the job a lot easier.


Thank you so much!

**Comment by F-Burning** [comment]
Here's an example of what I mean.

This drawing...


<img width="184" alt="0" src="https://user-images.githubusercontent.com/74098734/155198118-f4285f6a-7316-4529-a7da-0ecc9a93e555.png">



...is divided into two layers:


The hair...

<img width="184" alt="1" src="https://user-images.githubusercontent.com/74098734/155198066-1a6b9d13-21d1-4d56-95ee-c8dd8412eda8.png">


...and the head.

<img width="184" alt="2" src="https://user-images.githubusercontent.com/74098734/155198088-cbe24304-eed0-4db7-af33-9028b815f14e.png">


If I combine the two parts into a single layer and apply the smoothing to it, everything looks fine...

![01](https://user-images.githubusercontent.com/74098734/155198415-d8edbbc0-3e5c-45ed-85d7-03b733298aa1.jpg)

...but if I don't combine the layers and apply anti-aliasing individually to each one, the end result leaves "gaps" that make the red background behind it show through

![02](https://user-images.githubusercontent.com/74098734/155198566-548be9ad-5b52-4749-867b-31a140c6e9e9.jpg)

**Comment by artisteacher** [comment]
Would it be possible for the RGB Palette FX to have multiple sockets - like an OverFX so that this could be applied to as many columns as needed at once? Maybe the RGB palette filter could also include an erode/dilate setting to correct poor areas of overlap.

**Comment by F-Burning** [comment]
Honestly, for now I'm not very convinced by the second option proposed, because I see it as a partial solution that would depend on "patches" like what @artisteacher says about erode/dilate, and I'm not convinced by the idea of ​​creating an infinity of connectors depending on the complexity of the composition.

I think a simple node that smooths everything out, even though sub-xsheets need to be created, is a more practical option from my point of view.

PS: What if instead of using the camera, you relate all the columns to an assistant that is in charge of moving them? Would the same thing happen?

**Comment by F-Burning** [comment]
A suggestion from ignorance:

Couldn't the antialiasing node include an auto DPI option to match the camera?

I must admit that I don't know what I'm talking about, but to propose ideas 😅

**Comment by RodneyBaker** [comment]
This report relates to #3247

**Comment by F-Burning** [comment]
Thanks @RodneyBaker, I was unaware of that suggestion.

I have a question for @shun-iwasawa , I hope not to bother you.

When you apply an Fx after the 'scene' node, it's not like applying it after the camera work?

![Nuevo proyecto](https://user-images.githubusercontent.com/74098734/155858305-e7a8f083-3284-480d-924b-85abf82b1d43.png)

**Comment by F-Burning** [comment]
What about to apply anti-aliasing to sub-xsheets?

I mean, you do the compositing without camera work, then collapse all in a sub-xsheet, and apply the anti-aliasing to all is inside.

Actually you can't apply antialiasing to a sub-xsheet, but why the option is there?

![ppp](https://user-images.githubusercontent.com/74098734/158704929-28ac39bc-bfa0-4324-ac07-fec133ddf861.jpg)

**Comment by F-Burning** [comment]
Ok, I've deleted all the unnecessary post that I've made, and I'm going to expose clearly why an Fx Smooth is possible and why is necessary a node. 

The purpose of this node is for "advance compositing" with **RGB keying** like in After Effects, so, it's only for people that are interested in the spirit of learning or for professional work. If not, they have the level settings to apply the anti-aliasing.

- **Why is necessary a Smooth/Antialiasing node for anime complex compositing?**

In After Effects anime compositing, is common pre-composes anything, so, subxsheets are very important to make advance compositing. The point with this is subxsheets are not the problem, since they are necessary anyway.

- **An example with a character divided in different levels:**

If **A,B,C** and **D** are parts of the same character, but in different levels, each one with **4 frames**, it is more easy collapse them in one single column, and then make the RBG Keying work. 


![A1](https://user-images.githubusercontent.com/74098734/159127219-fe424bb4-0645-483b-9b60-60f5ffee0cfc.jpg)


![A2](https://user-images.githubusercontent.com/74098734/159127240-a5df1d04-32b7-4b99-9f53-6da6d608511c.jpg)


But why? Because in case you need to do changes in **A,B,C** and **D** leves, I can open the subxsheet and work with only 4 frames. Easy to see and easy to work.


Then I close the subxsheet, and I continue exposing the frames along the Xsheet/Timeline without worrying about its complexity with keyframes or anything. Once I make this, I apply the Antialiasing/Smooth node to the subxsheet/scene with the RGB work and without gaps between its parts, and collapses all one more time in a new subxsheet to make the camera work.



Here's another example with two different levels. Neither of them overlaps the other:

**- Applying antialiasing to each part individually with a green background.**


![A7](https://user-images.githubusercontent.com/74098734/159127660-b361c411-dfad-4445-859d-df0d59fb4988.jpg)


**- Applying antialiasing to all parts collapses (no gaps) with a green background.**


![A8](https://user-images.githubusercontent.com/74098734/159127665-786db41c-a48a-4a10-8f0a-b1457012e175.jpg)



In addition, collapses in Opentoonz are more efficient that pre-compose in After Effects, because subxsheets doesn't crop the material with camera size as pre-compose do, so you can do camera work with origin size levels, no matter how many times you collapse them.


**- There's one problem**

RBGKey doesn't work as I expected On aliases colors when I'm doing keying, because it leaves pixel artifacts:

![A9](https://user-images.githubusercontent.com/74098734/159127846-96c8d4c2-5b6c-4548-828f-c8cf7a08dab1.jpg)


Have this a solution? I'm thinking this could be related with the "blurry effect" witch you can see when turn on the preview with aliases pixels, but I have no idea. If this is unsolvable, the RGB keying work and the anti-aliasing node would not be possible.


And that's all. Something I would like to know, as far as possible and without pretending to demand anything, is to know whether or not we can expect this node on Opentoonz. I sincerely believe that it is a small request that can mean a big change.

Thanks for your attention.

**Comment by F-Burning** [comment]
> **- There's one problem**
> 
> RBGKey doesn't work as I expected On aliases colors when I'm doing keying, because it leaves pixel artifacts:
> 

Now I understand what's happens here. It was because I moved the camera before applying the antialiasing. 

In conclusion, there is no problems with RGBKey.

**Comment by F-Burning** [comment]
> Well, I'm thinking of two ideas, the Smooth fx and an alternative.
> 
> 1. Introducing the Smooth Fx
>    As I wrote before, usage of the Smooth Fx will be very limited. Because you cannot apply the fx to any level with camera work. Practical usage would be collapsing the level + smooth fx into a subxsheet and apply camera work to the subxsheet.
>    It seems to be somewhat consistent with the usage in AfterEffects, but I think it is a bit troublesome...
>    ![idea1_smoothfx](https://user-images.githubusercontent.com/17974955/155056164-a2f68470-8663-4b4d-bb56-be29d1197752.jpg)
> 2. Alternative idea: Raster Palette Filter Fx
>    This idea is premised on that you would like to do **RGB-based keying** before smoothing.
>    This effect would be specialized to use connecting just after the raster level column. It works like RGB Key Fx, but it will be applied BEFORE antialiased or resampled. (We already have this type of Fx for Toonz levels : the Palette Filter Fx.)
>    ![idea2_rasterpalettefilter](https://user-images.githubusercontent.com/17974955/155056276-118f80d6-0d62-43fa-9764-876a93cf3ed4.jpg)
> 
> I prefer the latter idea, as it can keep the scene simple.


First one remembers me a lot to After Effects with pre-componses. Also, you could work with other image formats like PNG or TGA and then apply the smooth.

Please, consider this. OT has a lot of japanese animation functions, but this one is missing and you cannot complete the task.

Thank you Shun.

**Comment by F-Burning** [comment]
Sorry to bother you @shun-iwasawa , but you were right. 


I didn't know how the filter palette worked, and yesterday I realized that the function was there all along 🤪 

I can work with anti-aliasing!



![Link1_1](https://user-images.githubusercontent.com/74098734/204790811-af3d0ae9-7841-454a-8942-2a19082929b8.png)
![Link2 0001](https://user-images.githubusercontent.com/74098734/204790823-70db29dd-9dea-4e2d-90cd-efe751d72172.png)


So the Raster palette filter would be perfect.

I am very sorry for the inconvenience, I am very ignorant on the subject, and I hope that in the future you can consider this node. Thank you very much!

**Comment by F-Burning** [comment]
Ok no, I still found issues with the smooth before the whole compositing, so I think is better add the option to sub-scenes.

**Comment by KaseyFarron** [comment]
I'd like to see this too, I always end up with artifacts when I have antialiasing enabled in the level options. For example here's a comparison from Natron where I make a very harsh adjustment to saturation and the image on the left is when I apply antialiasing node -after- all other effects, the image on the right is when I apply it -before- the effects, look at the eye white part.

![image](https://user-images.githubusercontent.com/55353161/220824919-b0f66d65-e9ad-42c0-a589-fd15b28f99cd.png)

It's the same with OpenToonz, I'm stuck with the artifacts.

**Comment by konero** [comment]
There is a **MLAA** shader provided by AMD here: https://github.com/GPUOpen-LibrariesAndSDKs/MLAA11

Which was ported as a GLSL shader here: https://www.shadertoy.com/view/cllXRB

Someone could port it to OpenToonz, then that way antialiasing could be handled by a node applying the effect as a texture? Unless this algorithm is no good for artwork.

**Comment by F-Burning** [comment]
Thanks @konero, it's another option.

I would like to know @manongjohn opinion on this topic, since he often does magic with Tahoma 2D with things that seem problematic in Opentoonz.

**Comment by manongjohn** [comment]
Thank you @F-Burning for the compliment. Unfortunately I do not have much expertise when it comes to creating Fx and how Fx are applied within OT.  I'd have to defer to @shun-iwasawa's expertise as to how something like this could be implemented.

General opinions: It does seem like a smoothing fx node would be a good thing to have, even if there are limitations on its usage.  That MLAA shader is pretty cool. :)

**Comment by konero** [comment]
There's an older After Effects plugin that was made open-source since 2017 (it used to be paid) called Loilo Smooth.

You can find it here: https://github.com/loilo-inc/smooth

Perhaps this could be ported? Shouldn't be any licencing issues. I think the node should completely rasterize its input (so it would no longer be possible to use Toonz Level nodes on its output) that just acts as a post processing node only.

![image](https://user-images.githubusercontent.com/19820721/221369663-c50157f4-0300-4e1f-b2d5-9d4a4fede25f.png)

**Comment by F-Burning** [comment]
Looks a good option too @konero 👍

I'd like to ask @shun-iwasawa if the anti-aliasing could work without camerawork, but using pegs instead or parenting levels.

**Comment by RodneyBaker** [comment]
Likely addressed via #6150
